lib/jnpr/junos/decorators.py

```diff
@@ -119,14 +119,20 @@ def wrapper(*args, **kwargs):
                             if isinstance(ignore_warn, (str, unicode)):
                                 if not re.search(ignore_warn, err['message'],
                                                  re.I):
+                                    # Message did not match.
                                     raise ex
                             elif isinstance(ignore_warn, list):
                                 for warn_msg in ignore_warn:
-                                    if not re.search(warn_msg, err['message'],
-                                                     re.I):
-                                        raise ex
+                                    if re.search(warn_msg, err['message'],
+                                                 re.I):
+                                        # Warning matches. Break skips else.
+                                        break
```

Comment:
In the case you just described, I would expect that the exception would be raised to the user because the first warning message (`Advertisement-interval is less than four`) did not match any of the `ignore_warning` values.

Stated another way, the normal exception is only suppressed if every warning message matches at least one of the values in `ignore_warning`.

I believe that's what my current code does. Here's a quick run down:
- If the exception doesn't have an `errs` attribute, raise the exception (line 142)
- Loop through every error/warning message (line 117):
    - If the message is not a warning, raise the exception (line 136)
    - If `ignore_warning` is a single string, and the message doesn't match, raise the exception (line 123)
    - If `ignore_warning` is a list of strings, loop through each string in the list:
        - Compare the message to the current string from `ignore_warning`
        - If the message matches the current string from `ignore_warning`, go to the next warning/error message
       - If the message didn't match any string in the `ignore_warning` list, then raise the exception (line 133)
- If all messages have been checked, then we know that all messages matched at least one of the values in `ignore_warning`, and therefore the exception should be suppressed and the rpc reply should be returned (line 138).

Again, let me know if this still doesn't make sense.